src/cswinrt/code_writers.h

```diff
@@ -3724,12 +3725,24 @@ R"(
         }
         else
         {
-            w.write(
+            if (get_category(type) == category::delegate_type)
+            {
+                w.write(
+R"(            
+            % unboxedValue = (%)ComInterfaceDispatch.GetInstance<object>((ComInterfaceDispatch*)thisPtr);  
+            var value = %Marshaller.ConvertToUnmanaged(unboxedValue).DetachThisPtrUnsafe();
```

Comment:
Nit: better to be explicit
```suggestion
            void* value = %Marshaller.ConvertToUnmanaged(unboxedValue).DetachThisPtrUnsafe();
```

Reply:
fixed